test/checkout.int.t.sol

```diff
@@ -1119,8 +1171,150 @@ contract Checkout_swapFromSupplierWithoutFeeSpecialOrder is Checkout {
     assertEq(removalAmounts[0], certificateAmount);
     assertEq(
       _bpNori.balanceOf(_namedAccounts.supplier),
-      (certificateAmount * customPriceMultiple) / 100 / 2 // divide to account for price multiple scale and then holdback percentage of 50%
+      (certificateAmount * customPriceMultiple) / 100 / 2 // divide to account for price multiple scale and holdback percentage of 50%
     );
     assertEq(_bpNori.balanceOf(_namedAccounts.feeWallet), 0);
   }
+
+  function test_revertsWhenSupplierDoesNotExistInMarket() external {
+    vm.prank(owner);
+    vm.expectRevert(InsufficientSupply.selector);
+    _market.swapWithoutFeeSpecialOrder({
+      recipient: owner,
+      purchaser: owner,
+      amount: certificateAmount,
+      customFee: customFee,
+      customPriceMultiple: customPriceMultiple,
+      supplier: _namedAccounts.supplier2, // not listed in market
+      vintages: new uint256[](0)
+    });
+  }
+}
+
+contract Checkout_swapWithoutFeeSpecialOrder_specificVintagesSpecificSupplier is
+  Checkout
+{
+  uint256 ownerPrivateKey = 0xA11CE;
+  address owner = vm.addr(ownerPrivateKey);
```

Comment:
Consider using _namedAccounts instead. It has the added benefit of labeling the account with "owner" due to how they are set up when looking at foundry logs. Doing it this way is just going to render the address with no label and because you don't have any indication what Alice's public key is here it'll be near impossible to debug. If you are doing this elsewhere in this PR change that too.

Reply:
This applies to multiple tests you've added to this file

Reply:
switched to using `_namedAccounts.buyer` everywhere where we don't need access to the private key for permit sigs

Reply:
https://book.getfoundry.sh/reference/forge-std/derive-remember-key?highlight=Derivekey#see-also

Additional "see also" links are also relevant

Reply:
very cool.  trying to figure out how that works with our `_namedAccounts` given that they don't use a mnemonic but `deriveKey` requires a mnemonic... unless the mnemonic is equivalent to `uint160(uint256((keccak256(abi.encodePacked(name)))))` which is being fed into `address`.  at any rate, did a huge test hygiene overhaul in this PR including a bunch of stuff that wasn't otherwise affected here so I'm gonna leave this one for another day

Reply:
Do we not use a mnemonic? Thought I had configured that, but maybe didn't get to it.

Reply:
no it looks like this: 

```
    NamedAccounts({
      // todo use .env: https://book.getfoundry.sh/tutorials/solidity-scripting?highlight=.env#deploying-our-contract
      deployer: 0x465d5a3fFeA4CD109043499Fa576c3E16f918463,
      _this: account("_this"),
      admin: account("admin"),
      supplier: account("supplier"),
      supplier2: account("supplier2"),
      supplier3: account("supplier3"),
      buyer: account("buyer"),
      feeWallet: account("feeWallet")
    });
```

where `account` looks like this:

```
  function account(string memory name) internal returns (address) {
    address addr = keccak256(bytes(name)) == keccak256("_this")
      ? address(this)
      : address(uint160(uint256((keccak256(abi.encodePacked(name))))));
    vm.label(addr, name);
    return addr;
  }
```